xpdo/om/mysql/xpdomanager.class.php

```diff
@@ -436,6 +445,13 @@ protected function getColumnDef($class, $name, $meta, array $options = array())
         if (empty ($extra) && isset ($meta['extra'])) {
             $extra= ' ' . $meta['extra'];
         }
+        $charset = '';
+        if (!empty($options['charset']) && in_array($this->xpdo->driver->getPhpType($dbtype), array('string'))) {
```

Comment:
Do we need to support other PHP types (e.g. `array`, which gets serialized automatically) here? Perhaps this ought to check the dbtype for a whitelist of column types that support character sets instead?